rclcpp/include/rclcpp/strategies/msg_pool_memory_strategy.hpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_RCLCPP_MSG_POOL_MEMORY_STRATEGY_HPP_
+#define RCLCPP_RCLCPP_MSG_POOL_MEMORY_STRATEGY_HPP_
+
+#include <rclcpp/msg_memory_strategy.hpp>
+#include <rclcpp/macros.hpp>
+
+namespace rclcpp
+{
+namespace msg_pool_memory_strategy
+{
+
+template<typename MessageT, size_t size>
```

Comment:
Since `size` is a constant within the template please make it camel case (`Size` like `MessageT`) to avoid collisions in the code.